tests/keystore.spec.ts

```diff
@@ -156,6 +156,80 @@ export default () => { describe('keystore', function() {
 
     });
 
+    describe('SharedKeyStore', function() {
+        const ks = new KeyStore(new Uint8Array(32).fill(0xff));
+        const sks = ks.getSharedKeyStore(ks.publicKeyBytes);
+
+        const nonce = new Uint8Array(24).fill(0xff);
+        const data = new Uint8Array(10).fill(0xff);
+
+        it('calculates the shared key', () => {
+            const key = hexToU8a('9cfcb55fa42de280c84c95d9cf08fcbec63657998d15e139dbd3b4c6a1264541');
+            expect((sks as any)._sharedKey).toEqual(key);
+        });
+
+        it('can be derived from a KeyStore', () => {
+            expect(sks.localSecretKeyBytes).toEqual(ks.secretKeyBytes);
+            expect(sks.localSecretKeyHex).toEqual(ks.secretKeyHex);
+            expect(sks.remotePublicKeyBytes).toEqual(ks.publicKeyBytes);
+            expect(sks.remotePublicKeyHex).toEqual(ks.publicKeyHex);
+        });
+
+        it('can be constructed from Uint8Array based keys', () => {
+            const sks = new SharedKeyStore(ks.secretKeyBytes, ks.publicKeyBytes);
+            expect(sks.localSecretKeyBytes).toEqual(ks.secretKeyBytes);
+            expect(sks.localSecretKeyHex).toEqual(ks.secretKeyHex);
+            expect(sks.remotePublicKeyBytes).toEqual(ks.publicKeyBytes);
+            expect(sks.remotePublicKeyHex).toEqual(ks.publicKeyHex);
+        });
+
+        it('can be constructed from hex string based keys', () => {
+            const sks = new SharedKeyStore(ks.secretKeyHex, ks.publicKeyHex);
+            expect(sks.localSecretKeyBytes).toEqual(ks.secretKeyBytes);
+            expect(sks.localSecretKeyHex).toEqual(ks.secretKeyHex);
+            expect(sks.remotePublicKeyBytes).toEqual(ks.publicKeyBytes);
+            expect(sks.remotePublicKeyHex).toEqual(ks.publicKeyHex);
+        });
+
+        it('rejects invalid keys', () => {
+            let create, error;
+
+            create = () => new SharedKeyStore({ meow: true }, ks.publicKeyBytes);
+            error = new ValidationError('Local private key must be an Uint8Array or a hex string');
+            expect(create).toThrow(error);
+
+            create = () => new SharedKeyStore(ks.secretKeyBytes, { meow: true });
+            error = new ValidationError('Remote public key must be an Uint8Array or a hex string');
+            expect(create).toThrow(error);
+        });
+
+        it('can encrypt and decrypt properly (round trip)', () => {
+            const expected = new Uint8Array(24).fill(0xee);
+            let encrypted;
+
+            encrypted = sks.encrypt(expected, nonce);
+            expect(sks.decrypt(encrypted)).toEqual(expected);
+            expect(sks.decryptRaw(encrypted.data, encrypted.nonce)).toEqual(expected);
+
+            encrypted = sks.encryptRaw(expected, nonce);
+            const encryptedBox = new Box(nonce, encrypted, nacl.box.nonceLength);
+            expect(sks.decrypt(encryptedBox)).toEqual(expected);
+            expect(sks.decryptRaw(encrypted, nonce)).toEqual(expected);
+        });
```

Comment:
You could also add a test where you encrypt/decrypt using the shared keystore and also with the regular keystore, and ensure that the results are the same :)

Reply:
Done!